modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

```diff
@@ -9,21 +9,82 @@ WARNING: On
 ifeval::["{project-context}" == "che"]
 {platforms-name} older than v1.13 and
 endif::[]
-OpenShift {ocp3-ver}, secrets mounted as file overrides volume mounts defined in the devfile.
+OpenShift {ocp3-ver}, secrets mounted as a file override volume mounts defined in the devfile.
 
-This section describes how to mount a secret from the user's {orch-namespace} as a file in single-workspace or multiple-workspace containers of {prod-short}.
+A secret can be mounted from the user's {orch-namespace} as a file in single-workspace or multiple-workspace containers of {prod-short}.
 
 .Prerequisites
 
-* A running instance of {prod-short}. To install an instance of {prod-short}, see {link-installing-an-instance}.
+* A running instance of {prod-short}.
++
+To install an instance of {prod-short}, see {link-installing-an-instance}.
 
 .Procedure
 
-. Create a new {platforms-name} secret in the {platforms-name} {orch-namespace} where a {prod-short} workspace will be created.
+To mount a secret as a file, complete the following steps as shown in the next example:
+
+. Create a new {platforms-name} secret in the {platforms-namespace} of the {prod-short} workspace.
+. Match the secret's `labels` to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
+. Define the default behavior for mounting secrets in the `automount-workspace-secret` annotation, which takes a boolean value:
+** `'true'` enables automatic mounting into all workspace containers.
+** `'false'` disables automatic mounting into all workspace containers. To mount a secret, a user must request it in the devfile component by using the `automountWorkspaceSecrets:true` property.
+. Set a `mount-path` for the file.
+. Include the `mount-as: file` annotation.
+. Add the file's name in the secret's `data` field.
+
+NOTE: A {orch-name} secret can contain several `data` items. The names of `data` items must match the name of the secret's file mounted into the container.
 
-* The labels of the secret that is about to be created must match the set of labels configured in `che.workspace.provision.secret.labels` property of {prod-short}. The default labels are:
+.Example: Mounting a secret as a file
+[source,yaml,subs="+quotes,attributes"]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mvn-settings-secret
+  labels:
+    app.kubernetes.io/part-of: che.eclipse.org <1>
+    app.kubernetes.io/component: workspace-secret <2>
+  annotations:
+    che.eclipse.org/automount-workspace-secret: 'true' <3>
+    che.eclipse.org/mount-path: {prod-home}/.m2/ <4>
+    che.eclipse.org/mount-as: file <5>
+data:
+  settings.xml: __<base64-encoded_value>__ <6>
+----
+<1> Default label.
+<2> Default label.
+<3> Enabled automatic mounting into all workspace containers.
+<4> Mount path, `{prod-home}/.m2/`, set for the file in all workspace containers.
+<5> Included `mount-as: file` annotation.
```

Comment:
```suggestion
<5> Mounts the secret as a file. 
```
Are the other options described in a separate reference document?

Reply:
@rolfedh Actually, all of these are described in the procedure steps that precede the example, and that's why I phrased the callouts this way. Also, I tried to make the structure of the callouts more or less parallel. This particular callout is phrased because of step 5 ("Include the `mount-as: file` annotation") in the procedure, which in the example becomes "Included `mount-as: file` annotation."

Reply:
Fixed. Thanks.